engine/src/witness/sol/sol_deposits.rs

```diff
@@ -1,204 +1,111 @@
-										deposit_channel.deposit_channel.address,
-										token_pubkey,
-									)
-									.expect("Failed to derive associated token account")
-									.address,
-									token_pubkey,
-									deposit_channel.deposit_channel.asset,
-								),
-							}
-						})
-						.collect::<Vec<_>>();
-
-					if !deposit_channels_info.is_empty() {
-						let chunked_deposit_channels_info = deposit_channels_info
-							.chunks(MAX_MULTIPLE_ACCOUNTS_QUERY / 2)
-							.map(|chunk| chunk.to_vec())
-							.collect::<Vec<Vec<_>>>();
-
-						for chunk_deposit_channels_info in chunked_deposit_channels_info {
-							let (ingresses, slot) = ingress_amounts(
-								&sol_rpc,
-								chunk_deposit_channels_info,
-								vault_address,
-							)
-							.await?;
-
-							if !ingresses.is_empty() {
-								let mut cached_balances = cached_balances.lock().await;
-
-								// Filter out the deposit channels that have the same balance
-								let new_ingresses: Vec<(DepositChannelType, u128)> = ingresses
-									.into_iter()
-									.filter_map(|(deposit_channel, amount)| {
-										let deposit_channel_cached_balance = cached_balances
-											.get(deposit_channel.address_to_witness())
-											.unwrap_or(&0);
-
-										if amount > *deposit_channel_cached_balance {
-											// TODO: This is a workaround for the current pallet.
-											// With the new pallet we could submit "amount" as is
-											// but we'd still need to keep track of the cached
-											// balance to avoid continous submissions.
-											// Some((deposit_channel, amount))
-											Some((
-												deposit_channel,
-												amount - deposit_channel_cached_balance,
-											))
-										} else {
-											None
-										}
-									})
-									.collect::<Vec<_>>();
-
-								if !new_ingresses.is_empty() {
-									process_call(
-										pallet_cf_ingress_egress::Call::<
-											_,
-											ChainInstanceFor<Inner::Chain>,
-										>::process_deposits {
-											deposit_witnesses: new_ingresses
-												.clone()
-												.into_iter()
-												.map(|(deposit_channel, amount)| {
-													pallet_cf_ingress_egress::DepositWitness {
-												deposit_address: *deposit_channel.address(),
-												asset: deposit_channel.asset(),
-												amount: amount
-													.try_into()
-													.expect("Ingress witness transfer value should fit u64"),
-												deposit_details: (),
-											}
-												})
-												.collect(),
-											block_height: slot,
-										}
-										.into(),
-										epoch.index,
-									)
-									.await;
-
-									// Update hashmap
-									new_ingresses.into_iter().for_each(
-										|(deposit_channel, value)| {
-											cached_balances.insert(
-												*deposit_channel.address_to_witness(),
-												value,
-											);
-										},
-									);
-								}
-							}
-						}
-					}
+const MAXIMUM_CONCURRNET_RPCS: usize = 16;
```

Comment:
Any particular reason why this value is set to 16?

Reply:
I don't believe there is - but seems a reasonable number.